java/memory/src/main/java/org/apache/arrow/memory/BufferAllocator.java

```diff
@@ -50,6 +50,27 @@
    */
   public ArrowBuf buffer(int size, BufferManager manager);
 
+  /**
+   * Allocate a new or reused buffer of the provided size. Note that the buffer may technically
```

Comment:
nit: Allocates (https://google.github.io/styleguide/javaguide.html#s7.2-summary-fragment)

Reply:
@emkornfield Thanks for your careful review. I have revised it accordingly.